### PR TITLE
Fixed bad redireccion of the link with "Storing Values with the Equal Operator"

### PR DESCRIPTION
**Pull request by frodrisan**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/generate-random-fractions-with-javascript.english.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ challengeType: 1
 <section id='description'>
 Random numbers are useful for creating random behavior.
 JavaScript has a <code>Math.random()</code> function that generates a random decimal number between <code>0</code> (inclusive) and not quite up to <code>1</code> (exclusive). Thus <code>Math.random()</code> can return a <code>0</code> but never quite return a <code>1</code>
-<strong>Note</strong><br>Like <a href='storing-values-with-the-assignment-operator' target='_blank'>Storing Values with the Equal Operator</a>, all function calls will be resolved before the <code>return</code> executes, so we can <code>return</code> the value of the <code>Math.random()</code> function.
+<strong>Note</strong><br>Like <a href='javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator' target='_blank'>Storing Values with the Equal Operator</a>, all function calls will be resolved before the <code>return</code> executes, so we can <code>return</code> the value of the <code>Math.random()</code> function.
```
Comment (thecodingaviator) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/generate-random-fractions-with-javascript.english.md]:
```suggestion
<strong>Note</strong><br>Like <a href='javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator' target='_blank'>Storing Values with the Assignment Operator</a>, all function calls will be resolved before the <code>return</code> executes, so we can <code>return</code> the value of the <code>Math.random()</code> function.
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]

**Comment by RandellDawson** [comment]
When I test the link you changed locally, I get a 404, because it attempts to link to:

> http://localhost:8000/javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator instead

instead of:
> http://localhost:8000/learn/javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator

@Bouncey When I click the `Preview custom 404 page`, it still does not go to where it should.  Is that because the link needs to be prefixed with `learn` or is this a known Gatsby bug that will not be present in production?

image

**Comment by thecodingaviator** [comment]
@RandellDawson I think that's because on local, the learn.freecodecamp.org page is served from localhost:8000/learn but on the web the correct link most definitely is https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator/

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/generate-random-fractions-with-javascript.english.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ challengeType: 1
 <section id='description'>
 Random numbers are useful for creating random behavior.
 JavaScript has a <code>Math.random()</code> function that generates a random decimal number between <code>0</code> (inclusive) and not quite up to <code>1</code> (exclusive). Thus <code>Math.random()</code> can return a <code>0</code> but never quite return a <code>1</code>
-<strong>Note</strong><br>Like <a href='storing-values-with-the-assignment-operator' target='_blank'>Storing Values with the Equal Operator</a>, all function calls will be resolved before the <code>return</code> executes, so we can <code>return</code> the value of the <code>Math.random()</code> function.
+<strong>Note</strong><br>Like <a href='javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator' target='_blank'>Storing Values with the Assignment Operator</a>, all function calls will be resolved before the <code>return</code> executes, so we can <code>return</code> the value of the <code>Math.random()</code> function.
```
Comment (Bouncey) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/generate-random-fractions-with-javascript.english.md]: the current `master` branch differs from what we have in production. We are trying to move things that are on subdomains in to subdirectories. Therefore, this `href` needs to be prefixed with `/learn`.

**Review by Bouncey — CHANGES_REQUESTED** [review]

**Comment by RandellDawson** [comment]
This will be fixed in another PR #35125 which fixes all such challenges.

**Pull request closed without merging**